packages/flutter_tools/lib/src/commands/migrate_start.dart

```diff
@@ -0,0 +1,213 @@
+      printCommandText('flutter migrate apply', logger);
+      logger.printStatus('You may also abandon the existing migration and start a new one with:');
+      printCommandText('flutter migrate abandon', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    if (await hasUncommittedChanges(project.directory.path, logger, migrateUtils)) {
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    List<SupportedPlatform>? platforms;
+    if (stringArg('platforms') != null) {
+      platforms = <SupportedPlatform>[];
+      for (String platformString in stringArg('platforms')!.split(',')) {
+        platformString = platformString.trim();
+        platforms.add(SupportedPlatform.values.firstWhere(
+          (SupportedPlatform val) => val.toString() == 'SupportedPlatform.$platformString'
+        ));
+      }
+    }
+
+    final MigrateResult? migrateResult = await computeMigration(
+      verbose: _verbose,
+      flutterProject: project,
+      baseAppPath: stringArg('base-app-directory'),
+      targetAppPath: stringArg('target-app-directory'),
+      baseRevision: stringArg('base-revision'),
+      targetRevision: stringArg('target-revision'),
+      deleteTempDirectories: boolArg('delete-temp-directories'),
+      platforms: platforms,
+      preferTwoWayMerge: boolArg('prefer-two-way-merge'),
+      allowFallbackBaseRevision: boolArg('allow-fallback-base-revision'),
+      fileSystem: fileSystem,
+      logger: logger,
+      migrateUtils: migrateUtils,
+    );
+    if (migrateResult == null) {
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    _deleteTempDirectories(
+      paths: <String>[],
+      directories: migrateResult.tempDirectories,
+    );
+
+    await writeStagingDir(migrateResult, logger, verbose: _verbose, flutterProject: project);
+
+    logger.printStatus('The migrate tool has staged proposed changes in the migrate working directory.\n');
+    logger.printStatus('Guided conflict resolution wizard:');
+    printCommandText('flutter migrate resolve-conflicts', logger);
+    logger.printStatus('Check the status and diffs of the migration with:');
+    printCommandText('flutter migrate status', logger);
+    logger.printStatus('Abandon the proposed migration with:');
+    printCommandText('flutter migrate abandon', logger);
+    logger.printStatus('Accept staged changes after resolving any merge conflicts with:');
+    printCommandText('flutter migrate apply', logger);
+
+    return const FlutterCommandResult(ExitStatus.success);
+  }
+
+  /// Deletes the files or directories at the provided paths.
+  void _deleteTempDirectories({List<String> paths = const <String>[], List<Directory> directories = const <Directory>[]}) {
+    for (final Directory d in directories) {
+      d.deleteSync(recursive: true);
+    }
+    for (final String p in paths) {
+      fileSystem.directory(p).deleteSync(recursive: true);
```

Comment:
Deletion is prone to failure esp. on Windows. When it fails we should not crash the tool, and print an error message instead directing the user to manually delete.